### ISSUE TITLE
Relations classes don't have select function

### ISSUE THREAD

**Issue by genesiscz** [nunomaduro/larastan#814]
- Larastan Version: newest
- `--level` used: 5

### Description

Relations don't seem to know about the select function.

```
  97     Call to an undefined method Illuminate\Database\Eloquent\Relations\HasMany::select().
  104    Call to an undefined method Illuminate\Database\Eloquent\Relations\BelongsTo::select().
  107    Call to an undefined method Illuminate\Database\Eloquent\Relations\BelongsTo::select().
```

### Laravel code where the issue was found

```php
PreparedGroup::with([
	"preparedGroupDetails" => static fn (HasMany $q) => $q->select([
		"id", "prepared_group_id", "ticket_code",
	],),
	"preparedGroupDetails.ticket" => static fn (BelongsTo $q) => $q->select(
		["id", "ticket_code", "race_count_left", "race_count_total"],
	),
	"preparedGroupDetails.driver" => static fn (BelongsTo $q) => $q->select(Driver::$selectMinimal)
],);
```

If it's something not obvious, I can for sure make a more thorough example.

**Comment by szepeviktor** [comment]
Related to  #804

**Comment by canvural** [comment]
Hi,

Yeah, this seems similar to #804. I explained [here](https://github.com/nunomaduro/larastan/issues/804#issuecomment-815791796) why this happens.

But this should also be solved as a side effect of #816 Feel free to try the master branch and see if it fixed it for you.
